### PR TITLE
*: Migrate from flatcar-linux to flatcar

### PR DESCRIPTION
**Pull request by krnowak**
This also includes one of the dependencies - locksmith, that also went through the same migration.

- [x] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -83,3 +83,12 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
 	sigs.k8s.io/yaml v1.2.0 // indirect
 )
+
+// This stuff here is to handle issues with updating locksmith - it
+// has the same stuff in its go.mod file.
```
Comment (invidian) [inline go.mod:88]: Hmm, I guess this is something we should look at. I'm definitely not a fan of adding replace directives as they don't propagate automatically for consumers and cause all kinds of problems.

Also the comment regarding the grpc is not true anymore as far as I know.

Comment (krnowak) [reply go.mod:88]: I'll try removing the grpc replacement. How do I get past the linter here that yells at me that replacements are not allowed?

The place to look at is locksmith. This thing looks bogus indeed.

Comment (invidian) [reply go.mod:88]: As only `github.com/flatcar/locksmith/pkg/timeutil` is used from locksmith, how about we just copy the used code and drop this dependency? I had plans to deal with this at some point anyway.

Comment (krnowak) [reply go.mod]: Yep, it looks like we have quite a picky linter here. Will have a look at it tomorrow.

Comment (invidian) [reply go.mod]: I can handle it too, since it's my decision to operate at this linter config here.

**Comment by krnowak** [comment]
Rebased on top of #172. will rebase it again when it's merged.

**Review by invidian — APPROVED** [review]
LGTM, thanks for taking care of this and patience @krnowak

**Comment by invidian** [comment]
I'll try to get a new release out this month including this change.

**Comment by krnowak** [comment]
> I'll try to get a new release out this month including this change.

No worries, we do not include FLUO in Flatcar images, so there's no time pressure from our side.

**Pull request merged**